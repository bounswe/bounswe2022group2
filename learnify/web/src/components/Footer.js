import * as React from 'react';
import './component_styles.css';

export default function Footer() {

    const year = new Date().getFullYear();

    return (
        <div className='footer-parent-layout'>
            <div className='footer-container'>
                <div className='footer-row'>
                    <div className='footer-column-l'>
                        <div className='footer-item-bold'>Learnify</div>
                        <div className='footer-item-reg'>is a learning space for everyone, where learning is <br></br> achieved through collaboration.</div>
                    </div>
                    <div className='footer-column-r'>
                        <div className='footer-item-bold'>Home</div>
                        <div className='footer-item-bold'>Categories</div>
                        <div className='footer-item-bold'>Notifications</div>
                    </div>
                    <div className='footer-column-r'>
                        <div className='footer-item-bold'>Profile</div>
                        <div className='footer-item-bold'>Contact</div>
                        <div className='footer-item-bold'>About</div>
                    </div>
                    <div className='footer-column-r'>
                        <div className='footer-item-bold'>My Learning Spaces</div>
                        <div className='footer-item-bold'>Popular Learning Spaces</div>
                        <div className='footer-item-bold'>Recommended Learning Spaces</div>
                    </div>
                    <div className='footer-column-r'>
                        {year} Â© Learnify Team. All rights reserved.
                    </div>
                </div>
            </div>
        </div>
    );
}