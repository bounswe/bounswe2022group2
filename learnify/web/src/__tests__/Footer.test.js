import React from "react";
import Footer from "./../components/Footer";
import { screen, render, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

describe("Footer", () => {
    test('should render component layout correctly', async () => {
        const component = renderer.create(<BrowserRouter><Footer /></BrowserRouter>).toJSON();
        expect(component).toMatchSnapshot();
    });
}    );

test('footer renders correctly', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const footer = screen.getByTestId(/footer/i);
    expect(footer).toBeInTheDocument();
}  );

test('footer renders all links correctly', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const homeLink = screen.getByText(/home/i);
    const categoriesLink = screen.getByText(/categories/i);
    const notificationsLink = screen.getByText(/notifications/i);
    const profileLink = screen.getByText(/profile/i);
    const contactLink = screen.getByText(/contact/i);
    const aboutLink = screen.getByText(/about/i);
    const myLearningSpacesLink = screen.getByText(/my learning spaces/i);
    const popularLearningSpacesLink = screen.getByText(/popular learning spaces/i);
    const recommendedLearningSpacesLink = screen.getByText(/recommended learning spaces/i);
    expect(homeLink).toBeInTheDocument();
    expect(categoriesLink).toBeInTheDocument();
    expect(notificationsLink).toBeInTheDocument();
    expect(profileLink).toBeInTheDocument();
    expect(contactLink).toBeInTheDocument();
    expect(aboutLink).toBeInTheDocument();
    expect(myLearningSpacesLink).toBeInTheDocument();
    expect(popularLearningSpacesLink).toBeInTheDocument();
    expect(recommendedLearningSpacesLink).toBeInTheDocument();
} );

test('footer home link renders correctly', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const homeLink = screen.getByText(/home/i);
    expect(homeLink).toBeInTheDocument();
} );

test('footer home link should be clickable', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const homeLink = screen.getByText(/home/i);
    fireEvent.click(homeLink);
    expect(homeLink).toBeInTheDocument();
} );

test('footer categories link renders correctly', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const categoriesLink = screen.getByText(/categories/i);
    expect(categoriesLink).toBeInTheDocument();
} );

test('footer categories link should be clickable', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const categoriesLink = screen.getByText(/categories/i);
    fireEvent.click(categoriesLink);
    expect(categoriesLink).toBeInTheDocument();
}   );

test('footer notifications link renders correctly', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const notificationsLink = screen.getByText(/notifications/i);
    expect(notificationsLink).toBeInTheDocument();
} );

test('footer notifications link should be clickable', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const notificationsLink = screen.getByText(/notifications/i);
    fireEvent.click(notificationsLink);
    expect(notificationsLink).toBeInTheDocument();
} );

test('footer profile link renders correctly', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const profileLink = screen.getByText(/profile/i);
    expect(profileLink).toBeInTheDocument();
}   );

test('footer profile link should be clickable', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const profileLink = screen.getByText(/profile/i);
    fireEvent.click(profileLink);
    expect(profileLink).toBeInTheDocument();
} );

test('footer contact link renders correctly', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const contactLink = screen.getByText(/contact/i);
    expect(contactLink).toBeInTheDocument();
} );

test('footer contact link should be clickable', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const contactLink = screen.getByText(/contact/i);
    fireEvent.click(contactLink);
    expect(contactLink).toBeInTheDocument();
}   );

test('footer about link renders correctly', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const aboutLink = screen.getByText(/about/i);
    expect(aboutLink).toBeInTheDocument();
}   );

test('footer about link should be clickable', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const aboutLink = screen.getByText(/about/i);
    fireEvent.click(aboutLink);
    expect(aboutLink).toBeInTheDocument();
}  );

test('footer my learning spaces link renders correctly', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const myLearningSpacesLink = screen.getByText(/my learning spaces/i);
    expect(myLearningSpacesLink).toBeInTheDocument();
} );

test('footer my learning spaces link should be clickable', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const myLearningSpacesLink = screen.getByText(/my learning spaces/i);
    fireEvent.click(myLearningSpacesLink);
    expect(myLearningSpacesLink).toBeInTheDocument();
} );

test('footer popular learning spaces link renders correctly', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const popularLearningSpacesLink = screen.getByText(/popular learning spaces/i);
    expect(popularLearningSpacesLink).toBeInTheDocument();
} );

test('footer popular learning spaces link should be clickable', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const popularLearningSpacesLink = screen.getByText(/popular learning spaces/i);
    fireEvent.click(popularLearningSpacesLink);
    expect(popularLearningSpacesLink).toBeInTheDocument();
} );

test('footer recommended learning spaces link renders correctly', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const recommendedLearningSpacesLink = screen.getByText(/recommended learning spaces/i);
    expect(recommendedLearningSpacesLink).toBeInTheDocument();
}   );

test('footer recommended learning spaces link should be clickable', async () => {
    render(<BrowserRouter><Footer /></BrowserRouter>);
    const recommendedLearningSpacesLink = screen.getByText(/recommended learning spaces/i);
    fireEvent.click(recommendedLearningSpacesLink);
    expect(recommendedLearningSpacesLink).toBeInTheDocument();
} );

