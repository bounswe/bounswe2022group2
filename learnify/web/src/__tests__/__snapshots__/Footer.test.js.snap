// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render component layout correctly 1`] = `
<div
  className="footer-parent-layout"
  data-testid="footer"
>
  <div
    className="footer-container"
  >
    <div
      className="footer-row"
    >
      <div
        className="footer-column-l"
      >
        <div
          className="footer-item-bold"
        >
          Learnify
        </div>
        <div
          className="footer-item-reg"
        >
          is a learning space for everyone, where learning is 
          <br />
           achieved through collaboration.
        </div>
      </div>
      <div
        className="footer-column-r"
      >
        <a
          className="footer-item-bold"
          href="/home"
        >
          Home
        </a>
        <a
          className="footer-item-bold"
          href="/categories"
        >
          Categories
        </a>
        <a
          className="footer-item-bold"
          href="/notifications"
        >
          Notifications
        </a>
      </div>
      <div
        className="footer-column-r"
      >
        <a
          className="footer-item-bold"
          href="/profile/null"
        >
          Profile
        </a>
        <a
          className="footer-item-bold"
          href="/contact"
        >
          Contact
        </a>
        <a
          className="footer-item-bold"
          href="/about"
        >
          About
        </a>
      </div>
      <div
        className="footer-column-r"
      >
        <a
          className="footer-item-bold"
          href="/null/learningspaces"
        >
          My Learning Spaces
        </a>
        <a
          className="footer-item-bold"
          href="/popular"
        >
          Popular Learning Spaces
        </a>
        <a
          className="footer-item-bold"
          href="/recommended"
        >
          Recommended Learning Spaces
        </a>
      </div>
      <div
        className="footer-column-r"
      >
        2023
         Â© Learnify Team. All rights reserved.
      </div>
    </div>
  </div>
</div>
`;
